### PR TITLE
Documenting public APIs

### PR DESCRIPTION
**Pull request by isoos**
#2142

### CODE REVIEW

File: doc/api.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  "packages": [
+    "http",
+    "provider",
+    /* further package names */
+  ]
+}
+```
+
+The API returns the top package names on `pub.dev`. At the moment we
+limit the result length at the maximum of 20000 package names, which
```
Comment (sigurdm) [inline doc/api.md:42]:
```suggestion
limit the result length at the maximum of 20000 package names, this
```

File: doc/api.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+limit the result length at the maximum of 20000 package names, this
+limit may change in the future.
+
+The response is always a gzip-ed JSON content, and should be cached
+on the client side  for at least 8 hours between requests (as indicated
+by the `cache-control` header).
+
+Notes:
+ * Not all package names are included in this response.
+ * The inclusion criteria used by `pub.dev` may change without any early notice.
```
Comment (jonasfj) [inline doc/api.md:51]:
```suggestion
The API returns the top package names on `pub.dev`. To reduce payload size the
result may not include all package names. The size limitation is subject to change.

The response is always a gzip-ed JSON content, and should be cached
on the client side  for at least 8 hours between requests (as indicated
by the `cache-control` header).

Notes:
 * Not all package names are included in this response.
 * The inclusion criteria used by `pub.dev` may change without notice.
```

File: doc/api.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
+# `pub.dev` API for developers
+
+## Stay informed
+
+This document describes the officially supported API of the `pub.dev` site.
+`pub.dev` may expose API endpoints that are available publicly, but unless
+they are documented here, we don't consider them as officially
+supported, and may change or remove them without any notice.
```
Comment (jonasfj) [inline doc/api.md:8]:
```suggestion
supported, and may change or remove them without notice.
```

**Review by jonasfj — APPROVED** [review]

File: doc/api.md (hunk 2 above)
Comment (jonasfj) [reply doc/api.md:51]: Or something like htat... I would suggest avoiding terms like "We"...  And avoid being too specific in describing what the API does.

**Pull request merged**